kernel/console.c

```diff
@@ -116,7 +116,7 @@ static void console_bh_handler(uint8_t events)
 	if (events == BH_POLLIRQ) {
 		console_output_handler();
 		console_input_handler();
-	} else if (events == CONSOLE_IRQ) {
+	} else if ((events | console_events) == CONSOLE_IRQ) {
```

Comment:
Architecturally, we have only BH_WAKEUP and BH_POLLIRQ events, see include/target/bh.h.
So you can simplify the logic with:
else {
  if (console_events == CONSOLE_IRQ) {
    ...
  }
}